opencxl/util/number.py

```diff
@@ -108,6 +109,16 @@ def tlptoh64(n):
     return bswap64(n)
 
 
+def split_cacheline(
+    cacheline: int, line_length: int = 64, stride: int = 8
+) -> Generator[int, int, None]:
+    for unit in range(line_length - stride, -1, -stride):
+        bmask = ~((1 << unit) - 1)
+        masked = cacheline & bmask
+        yield masked >> unit
+        cacheline -= masked
```

Comment:
Doesn't return anything, but is used for assignment when called.

Reply:
This actually returns a python generator!

Reply:
I see. Just read about it (https://realpython.com/introduction-to-python-generators).

While I appreciate the functionality and have tried it myself:
```
from typing import Generator
def return_n_values(n):
    num = 0
    while num < n:
        yield num
        num += 1

a, b, c = return_n_values(3)
print(a, b, c)
```

I don't see a tangible benefit here. i.e. manipulate a dataset that doesn't fit in the memory. This is masking and assigning two values to two integers that could be used later. You could create an equivalent functionality by returning a tuple / list which would make the code more readable (albeit admittedly not as `cool`) :wink:. Please "simplify" this for the masses.

Reply:
Fair enough, but I do use the function elsewhere to split up large chunks of data into cachelines (e.g. `split_cacheline(data, len(data), 64)`). I spent a bit of time profiling this function call on large strings (ex. 80KB); not only was the generator slightly faster than the list solution, but it was also much more efficient in terms of memory usage.

Could I keep this function, but rename it to something more appropriate? Something like `split_int`, for example.

Reply:
Alright. Sounds good.